### PR TITLE
Refactor DAO and improve SQL queries

### PR DESCRIPTION
**Pull request by scan-lan**
just neatening

### CODE REVIEW

File: queries/allCountries.sql [hunk 1]
```diff
@@ -1,11 +1,5 @@
...
-    SELECT name
-    FROM city ci
-    WHERE countrycode = co.code
-        AND ci.id = co.capital
-    ) AS capital
-FROM country co
-ORDER BY population DESC;
-
-
+SELECT code, country.name, region, country.population, city.name
```
Comment (Craig180885-napier) [inline queries/allCountries.sql]: Does this need "As Capital" after city.name?

File: queries/topNCountries.sql [hunk 2]
```diff
@@ -1,34 +1,25 @@
-# The top N populated countries in the world where N is provided by the user.
-SELECT code, name, continent, region, population, (
-    SELECT name
-    FROM city ci
-    WHERE countrycode = co.code
-        AND ci.id = co.capital
-    ) AS capital
-FROM country co
+# The top N most populous countries in the world where N is provided by the user.
+SELECT country.code, country.name, continent, region, country.population, city.name AS capital
```
Comment (Craig180885-napier) [inline queries/topNCountries.sql:2]: inconsistency in the SELECT portion of the statement. i.e. "country.name, continent" would expect the same convesntion throughout.  My prefence would be tablename.coloumn but I'm not precious about it.

File: queries/topNCountries.sql [hunk 3]
```diff
@@ -1,34 +1,25 @@
...
-SELECT code, name, continent, region, population, (
-    SELECT name
-    FROM city ci
-    WHERE countrycode = co.code
-      AND ci.id = co.capital
-) AS capital
-FROM country co
+# The top N most populous countries in a continent where N is provided by the user.
+SELECT country.code, country.name, continent, region, country.population, city.name AS capital
+FROM country
```
Comment (Craig180885-napier) [inline queries/topNCountries.sql:11]: Same as above comment

File: queries/topNCountries.sql [hunk 4]
```diff
@@ -1,34 +1,25 @@
...
-SELECT code, name, continent, region, population, (
-    SELECT name
-    FROM city ci
-    WHERE countrycode = co.code
-      AND ci.id = co.capital
-) AS capital
-FROM country co
+# The top N most populous countries in a region where N is provided by the user.
+SELECT country.code, country.name, continent, region, country.population, city.name AS capital
+FROM country
```
Comment (Craig180885-napier) [inline queries/topNCountries.sql:20]: same as above comment

**Review by Craig180885-napier — COMMENTED** [review]
Just a few small points

File: queries/allCountries.sql (hunk 1 above)
Comment (Craig180885-napier) [reply queries/allCountries.sql]: same as below here as well

File: queries/topNCountries.sql (hunk 2 above)
Comment (scan-lan) [reply queries/topNCountries.sql:2]: the table.filedname is only necessary if there are fields with the same names in the tables that you're joining. example:
country has fields name, continent and population
city has fields name, district and population

when referring to `continent` you don't need the `country.` prefix as there is only one field called continent in the query, but when referring to `name` you have to specify whether it's the name in the city table or the name in the country table. if you don't specify that then you get SQL errors

**Review by Craig180885-napier — DISMISSED** [review]

**Review by LukasP23 — APPROVED** [review]

**Pull request merged**